web_src/js/features/repo-migrate.js

```diff
@@ -53,3 +56,14 @@ export function initRepoMigrationStatusChecker() {
 
   syncTaskStatus(); // no await
 }
+
+async function doRetry(e) {
+  await fetch($(e.target).attr('task-retry-url'), {
+    method: 'post',
+    headers: {
+      'X-Csrf-Token': csrfToken,
+      'Content-Type': 'application/json',
+    },
+  });
+  window.location.reload();
+}
```

Comment:
Limit this post api with the help of retry counter. Otherwise user will abuse this functionality.

Reply:
I don't think there is a need to add a limit on the number of retries here, because this is not an automated behavior, its retry action is triggered by the user's initiative, 
Therefore, if you add a limit to the number of times here, it is unfriendly and incomprehensible to the user.

Reply:
For example, like me (in China), for well-known reasons, access to github can be slow and unstable, and often times out; 
Sometimes I try three or five times and I succeed. 
Sometimes, it takes more than a dozen retries to succeed; 
Sometimes it takes a while to change the time period and then try again to succeed; 
In this case, how many times is it reasonable to limit it? 
Does retry operation cost more because of my network? (Because I have tried too many times, I am not allowed to try again. I can only delete and then re-create and continue to try again, which increases the operation cost in vain)

Reply:
As I said earlier, this action is not an automatic action, it is triggered by the user's subjective reasons. Increasing the number of retries is ultimately a compromise for Gitea, not a compromise for Gitea to provide a better experience for users;

Reply:
Thanks for the explanation. I got it.